src/charm.py

```diff
@@ -160,7 +155,8 @@ def _get_upf_host_port_list(self) -> List[Tuple[str, int]]:
         for fiveg_n4_relation in self.model.relations.get(FIVEG_N4_RELATION_NAME, []):
             if not fiveg_n4_relation.app:
                 logger.warning(
-                    "Application missing from the %s relation data", FIVEG_N4_RELATION_NAME
+                    "Application missing from the %s relation data",
+                    FIVEG_N4_RELATION_NAME,
                 )
                 return []
```

Comment:
in here we should continue instead of returning. otherwise all the other relations won't taken into account